server/models/Transaction.js

```diff
@@ -754,14 +759,85 @@ function defineModel() {
       data,
     };
 
-    await Transaction.createDoubleEntry(hostFeeTransaction);
+    const hostFeeTransaction = await Transaction.createDoubleEntry(hostFeeTransactionData);
 
     // Reset the original host fee because we're now accounting for this value in a separate set of transactions
     transaction.hostFeeInHostCurrency = 0;
 
     return { transaction, hostFeeTransaction };
   };
 
+  Transaction.createHostFeeShareTransactions = async (hostFeeTransaction, host, isDirectlyCollected = false) => {
+    const plan = await host.getPlan();
+    if (!plan.hostFeeSharePercent) {
+      return;
+    }
+
+    const amount = Math.round((hostFeeTransaction.amount * plan.hostFeeSharePercent) / 100);
+    const currency = hostFeeTransaction.currency;
+
+    const hostCurrency = HOST_FEE_SHARE_TRANSACTION_PROPERTIES.hostCurrency;
+    const hostCurrencyFxRate = await Transaction.getFxRate(currency, hostCurrency, hostFeeTransaction);
+    const amountInHostCurrency = Math.round(amount * hostCurrencyFxRate);
+
+    const hostFeeShareTransactionData = {
+      type: TransactionTypes.CREDIT,
+      kind: TransactionKind.HOST_FEE_SHARE,
+      description: 'Host Fee Share',
+      TransactionGroup: hostFeeTransaction.TransactionGroup,
+      FromCollectiveId: host.id,
+      CollectiveId: HOST_FEE_SHARE_TRANSACTION_PROPERTIES.CollectiveId,
+      HostCollectiveId: HOST_FEE_SHARE_TRANSACTION_PROPERTIES.HostCollectiveId,
+      // Compute amounts
+      amount,
+      netAmountInCollectiveCurrency: amount,
+      currency,
+      amountInHostCurrency,
+      hostCurrency,
+      hostCurrencyFxRate,
+      // No fees
+      platformFeeInHostCurrency: 0,
+      hostFeeInHostCurrency: 0,
+      paymentProcessorFeeInHostCurrency: 0,
+      OrderId: hostFeeTransaction.OrderId,
+      createdAt: hostFeeTransaction.createdAt,
+    };
+
+    const hostFeeShareTransaction = await Transaction.createDoubleEntry(hostFeeShareTransactionData);
+
+    let hostFeeShareDebtTransaction;
+    if (!isDirectlyCollected) {
+      hostFeeShareDebtTransaction = await Transaction.creatHostFeeShareDebtTransactions(
+        hostFeeShareTransactionData,
+        host,
+      );
+    }
+
+    return { hostFeeShareTransaction, hostFeeShareDebtTransaction };
+  };
+
+  Transaction.creatHostFeeShareDebtTransactions = async (hostFeeShareTransactionData, host) => {
+    // Create debt transaction
+    const hostFeeShareDebtTransactionData = {
+      ...omit(hostFeeShareTransactionData, ['data']), // TODO: remove the OrderId?
```

Comment:
The more I think about it, the more I find it useful to be able to group all transactions (including refunds) by `OrderId`